test/test_driver/test_sq_store.py

```
@@ -5,13 +5,59 @@
 import tempfile
 
 from squirrel.driver import StoreDriver
-from squirrel.iterstream import FilePathGenerator
+from squirrel.iterstream import FilePathGenerator, IterableSource
 from squirrel.serialization import MessagepackSerializer
 from squirrel.store import SquirrelStore
 from squirrel.fsspec.fs import get_fs_from_url
 from squirrel.integration_test.helpers import SHAPE, get_sample
 
 
+@pytest.mark.parametrize("exist_ok_while_clean", [True, False], "clean_while_exist_ok", [True, False])
+def test_squirrel_overwrite_when_creating(
```

Should we throw an error when the url is non-empty and add the `exist_ok` flag during creation?